DxDispatch/src/dxdispatch/Device.cpp

```diff
@@ -94,6 +94,8 @@ Device::Device(IAdapter* adapter, bool debugLayersEnabled, D3D12_COMMAND_LIST_TY
 
 Device::~Device()
 {
+    // Intentionally leak d3d12 module to avoid order-of-destruction issues with PIX wrapper.
```

Comment:
>// Intentionally leak d3d12 module to avoid order-of-destruction issues with PIX wrapper. [](http://example.com/codeflow?start=4&length=89)

😢 Granted, this is a special purpose tool that is pretty tied to D3D anyway, but in larger apps like Photoshop and Corel which load and unload DLLs, we want proper cleanup, and I worry about people looking at our samples and tools as inspiration for the right thing to do. Is there any way we can reduce this? e.g. Only when we use PIX?

Reply:
Yeah, this annoyed me as well. I think I misdiagnosed the issue, though, and it was actually an order-of-destruction issue with members of the Device class. It's cleaner to more explicitly manage the lifetime of these modules in the main function than to rely on member order.